compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -960,10 +970,27 @@ fn report_unreachable_pattern<'p, 'tcx>(
 }
 
 /// Report unreachable arms, if any.
-fn report_arm_reachability<'p, 'tcx>(cx: &PatCtxt<'p, 'tcx>, report: &UsefulnessReport<'p, 'tcx>) {
+fn report_arm_reachability<'p, 'tcx>(
+    cx: &PatCtxt<'p, 'tcx>,
+    report: &UsefulnessReport<'p, 'tcx>,
+    suggest_removing_if_unreachable: bool,
+) {
+    let sm = cx.tcx.sess.source_map();
     for (arm, is_useful) in report.arm_usefulness.iter() {
         if let Usefulness::Redundant(explanation) = is_useful {
-            report_unreachable_pattern(cx, arm.arm_data, arm.pat, explanation)
+            let (hid_id, arm_span) = arm.arm_data;
```

Comment:
We shouldn't need to (necessarily) string through the arm span, that should be recoverable from the `HirId`, no? Isn't it literally just `tcx.hir().span(hir_id)`?

Reply:
TIL!